components/renku_data_services/data_connectors/db.py

```diff
@@ -0,0 +1,225 @@
+    ) -> models.DataConnector:
+        """Update a data connector entry."""
+        if not session:
+            raise errors.ProgrammingError(message="A database session is required.")
+        result = await session.scalars(
+            select(schemas.DataConnectorORM).where(schemas.DataConnectorORM.id == data_connector_id)
+        )
+        data_connector = result.one_or_none()
+        if data_connector is None:
+            raise errors.MissingResourceError(
+                message=f"Data connector with id '{data_connector_id}' does not exist or you do not have access to it."
+            )
+
+        current_etag = data_connector.dump().etag
+        if current_etag != etag:
+            raise errors.ConflictError(message=f"Current ETag is {current_etag}, not {etag}.")
+
+        # TODO: handle namespace or slug update
+        if patch.name is not None:
+            data_connector.name = patch.name
+        if patch.visibility is not None:
+            visibility_orm = (
+                apispec.Visibility(patch.visibility)
+                if isinstance(patch.visibility, str)
+                else apispec.Visibility(patch.visibility.value)
+            )
+            data_connector.visibility = visibility_orm
+        if patch.description is not None:
+            data_connector.description = patch.description if patch.description else None
+        if patch.keywords is not None:
+            data_connector.keywords = patch.keywords if patch.keywords else None
+        if patch.storage is not None:
+            if patch.storage.configuration is not None:
+                data_connector.configuration = patch.storage.configuration
+                data_connector.storage_type = data_connector.configuration["type"]
+            if patch.storage.source_path is not None:
+                data_connector.source_path = patch.storage.source_path
+            if patch.storage.target_path is not None:
+                data_connector.target_path = patch.storage.target_path
+            if patch.storage.readonly is not None:
+                data_connector.readonly = patch.storage.readonly
+
+        await session.flush()
+        await session.refresh(data_connector)
+
+        return data_connector.dump()
+
+    @with_db_transaction
+    async def delete_data_connector(
+        self,
+        data_connector_id: ULID,
+        *,
+        session: AsyncSession | None = None,
+    ) -> models.DataConnector | None:
+        """Delete a data connector."""
+        if not session:
+            raise errors.ProgrammingError(message="A database session is required.")
+        result = await session.scalars(
+            select(schemas.DataConnectorORM).where(schemas.DataConnectorORM.id == data_connector_id)
+        )
+        data_connector = result.one_or_none()
+        if data_connector is None:
+            return None
+
+        await session.delete(data_connector)
+
+        return data_connector.dump()
```

Comment:
issue: The only reason to return something here is if there's a decorator that uses the return value to send a delete message to the message queue, which isn't the case here. So a return value isn't needed.
Additionally, calling `dump()` after `delete()` can cause issues if there's lazy loaded child entities, as dump will try to load the child entities that were just deleted. This isn't an issue here, I think, but in general I started just returning the entity id in delete methods to prevent that problem and since delete messages only contain an id anyways.
So I'd either remove the return or return just the id.

Reply:
I think messages will have to be implemented in the future (post-build), but I'll remove the `dump()` and simply return `None` or the ID.

Reply:
Done in ceb40df67d2a0a519b334089c541d2a72453ef56.